File: app/src/main/java/com/example/notificationapp/data/network/api/RetrofitAccessObject.java
package com.example.notificationapp.data.network.api;

import com.example.notificationapp.utils.Constants;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitAccessObject {

    private static Api api;

    private RetrofitAccessObject() {
    }

    public static Api getRetrofitAccessObject() {
        if (api == null) {
            Gson gson = new GsonBuilder()
                    .setLenient()
                    .create();
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(Constants.BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create(gson))
                    .build();
            api = retrofit.create(Api.class);
        }
        return api;
    }
}

File: app/src/main/java/com/example/notificationapp/data/network/api/RetrofitAccessObject.kt
package com.example.notificationapp.data.network.api

import com.example.notificationapp.Constants
import com.google.gson.GsonBuilder
import okhttp3.OkHttpClient
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit

object RetrofitAccessObject {

    private var mApi: Api? = null
    val clientSetup = OkHttpClient.Builder()
        .connectTimeout(1, TimeUnit.MINUTES)
        .writeTimeout(1, TimeUnit.MINUTES) // write timeout
        .readTimeout(1, TimeUnit.MINUTES) // read timeout
        .build()

    fun getRetrofitAccessObject(): Api {
        if (mApi == null) {
            val gson = GsonBuilder().setLenient().create()
            val retrofit = Retrofit.Builder()
                .baseUrl(Constants.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create(gson))
                .client(clientSetup)
                .build()
            mApi = retrofit.create(Api::class.java)
        }
        return mApi!!
    }
}